embassy-executor/src/arch/std.rs

```diff
@@ -55,11 +55,24 @@ mod thread {
         ///
         /// This function never returns.
         pub fn run(&'static mut self, init: impl FnOnce(Spawner)) -> ! {
+            self.run_until(init, || false);
+            panic!("It should not be possible to get here.")
```

Comment:
the compiler can check this for you. remove this line and the `;` in the previous line and it'll compile.

Reply:
ah right `run_until()` is not `-> !`. Sorry, my bad, melted brain today.

so just one minor nit left: The idiomatic way to mark spots that should never be reached is `unreachable!()` then, not a panic with a message.